vcd/resource_vcd_vapp_vm.go

```diff
@@ -1042,6 +1157,112 @@ func updateStateOfAttachedDisks(d *schema.ResourceData, vm govcd.VM, vdc *govcd.
 	return d.Set("disk", transformed)
 }
 
+func updateStateOfInternalDisks(d *schema.ResourceData, vm govcd.VM) error {
+	err := vm.Refresh()
+	if err != nil {
+		return err
+	}
+
+	existingInternalDisks := vm.VM.VmSpecSection.DiskSection.DiskSettings
```

Comment:
Please add a comment why VmSpecSection is used here as opposed to the VirtualHardwareSection in getVmIndependentDisks function, and please add a similar comment in the getVmIndependentDisks function too. Imagine you're reading this file after a break, the fact that independent and internal disks are being read from different sections **with filters** is confusing, as, looking at code, it looks like both of these sections contain same both types of disks. At least, VmSpecSection is holding both types for sure. So, it's not clear why we aren't using the same list (section) in both places. I understand there's a reason which you know of, but please clarify that with a comment to make it easier for anyone reading this, as it is not intuitive.

Reply:
done

Reply:
The comments are enlightening. Thank you for adding!